patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch

```diff
@@ -0,0 +1,691 @@
+--- a/target/linux/ipq40xx/base-files/etc/board.d/01_leds
++++ b/target/linux/ipq40xx/base-files/etc/board.d/01_leds
+@@ -64,6 +64,11 @@ zyxel,wre6606)
+ 	ucidef_set_led_wlan "wlan2g" "WLAN2G" "${boardname}:green:wlan2g" "phy0tpt"
+ 	ucidef_set_led_wlan "wlan5g" "WLAN5G" "${boardname}:green:wlan5g" "phy1tpt"
+ 	;;
++cig,wf610d)
++	ucidef_set_led_wlan "wlan2g" "WLAN2G" "${boardname}:blue:wlan2g" "phy0tpt"
++	ucidef_set_led_wlan "wlan5g" "WLAN5G" "${boardname}:blue:wlan5g" "phy1tpt"
++	;;
++
+ *)
+ 	;;
+ esac
+diff --git a/target/linux/ipq40xx/base-files/etc/board.d/02_network b/target/linux/ipq40xx/base-files/etc/board.d/02_network
+index 56de6e61c3..33cc71296e 100755
+--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
++++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
+@@ -69,6 +69,11 @@ ipq40xx_setup_interfaces()
+ 		ucidef_add_switch "switch0" \
+ 			"0u@eth0" "3:lan" "4:lan" "0u@eth1" "5:wan"
+ 		;;
++	cig,wf610d)
++		ucidef_set_interfaces_lan_wan "eth0" "eth1"
++		ucidef_add_switch "switch0" \
++			"0u@eth0" "1:lan" "2:lan" "3:lan" "4:lan"
++
+ 	*)
+ 		echo "Unsupported hardware. Network interfaces not initialized"
+ 		;;
+diff --git a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+index ff77aed304..d36fc505ff 100644
+--- a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
++++ b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+@@ -199,6 +199,15 @@ case "$FIRMWARE" in
+ 	tp-link,ec420-g1)
+ 		ath10kcal_extract "0:ART" 4096 12064
+ 		ath10kcal_patch_mac_crc $(macaddr_add "$(get_tip mac_address)" 9) #2.4G, wlan1
++		;;
++	cig,wf610d)
++		mtd=$(find_mtd_chardev "0:ART")
++		[ -n "$mtd" ] || \
++			ath10kcal_die "no mtd device found for partition 0:ART"
++
++		dd if=$mtd of=/lib/firmware/$FIRMWARE bs=32 skip=128 count=377 2>/dev/null || \
++		ath10kcal_die "failed to extract calibration data from $mtd"
++
+ 		;;
+ 	esac
+ 	;;
+@@ -264,8 +273,17 @@ case "$FIRMWARE" in
+ 		ath10kcal_extract "ART" 20480 12064
+ 		ath10kcal_patch_mac_crc $(macaddr_add $(cat /sys/class/net/eth0/address) -1)
+ 		;;
++	cig,wf610d)
++		mtd=$(find_mtd_chardev "0:ART")
++		[ -n "$mtd" ] || \
++			ath10kcal_die "no mtd device found for partition 0:ART"
++
++		dd if=$mtd of=/lib/firmware/$FIRMWARE bs=32 skip=640 count=377 2>/dev/null || \
++		ath10kcal_die "failed to extract calibration data from $mtd"
++		;;
+ 	esac
+ 	;;
++
+ *)
+ 	exit 1
+ 	;;
+diff --git a/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh b/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
+index f8be25d8ff..29c4b50d6a 100644
+--- a/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
++++ b/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
+@@ -1,3 +1,4 @@
++. /lib/functions/system.sh
+ PART_NAME=firmware
+ REQUIRE_IMAGE_METADATA=1
+ 
+@@ -44,6 +45,20 @@ zyxel_do_upgrade() {
+ 	fi
+ }
+ 
++cig_do_upgrade() {
```

Comment:
also here, There is an exisitng code path to flash these units. please make use of it

Reply:
Hi, the only relevant path I find is nand_upgrade_tar, but it searches for Ubi partitions. which we don't have.